docroot/profiles/custom/sitenow/sitenow.profile

```diff
@@ -1049,19 +1038,21 @@ function sitenow_entity_insert(EntityInterface $entity) {
     // Load the node and grab the layout information.
     $node = \Drupal::service('entity.repository')
       ->loadEntityByUuid('node', $entity->uuid());
-    $layouts = $node->get('layout_builder__layout');
-
-    foreach ($layouts as $layout) {
-      $section = $layout->getValue()['section'];
-      // Pull out individual components.
-      foreach ($section->getComponents() as $component) {
-        // Grab the associated block's uuid.
-        $config = $component->get('configuration');
-        if (isset($config['block_revision_id'])) {
-          $rev_id = $config['block_revision_id'];
-          $block = $block_controller->loadRevision($rev_id);
-          if ($block) {
-            $use_controller->addUsage($block->id(), $node);
+    if ($node instanceof NodeInterface) {
+      $layouts = $node->get('layout_builder__layout');
+
+      foreach ($layouts as $layout) {
+        $section = $layout->getValue()['section'];
+        // Pull out individual components.
+        foreach ($section->getComponents() as $component) {
```

Comment:
Could we update this as well:
```
$layout = $node->get('layout_builder__layout');
foreach ($layout->getSections() as $section) {
...
}
```
I think it'd help with readability, and get rid of the awkward `$layout->getValue()['section']` (I don't have a clean branch, and also don't want to futz around and break things while you're all working!)

Reply:
I think I got this refactored, and working with the updated tests. It started complaining about the `$component->get('configuration')` further down, and I don't think the initial path we started on with `getThirdPartySettings()` was going to work, because we specifically needed the config. Workaround was to get the component's plugin, check that the plugin was an instance of an InlineBlock, and get the configuration from there.

I did also test with a fresh install locally, and blocks' media entities are editable, so I think this bit of rogue refactoring is good to go.